test/controllers/posts_controller_test.rb

```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+
+class PostsControllerTest < ActionDispatch::IntegrationTest
+  test 'should get index' do
+    get posts_index_url
+    assert_response :success
+  end
+
+  test 'should get show' do
+    get posts_show_url
+    assert_response :success
+  end
+end
```

Comment:
- [ ] These files should be under `spec/requests` directory to run the test not inside the `test` file.
- [ ] Here, I think you should use `type: request` as I have shown in the example code.

Example:

```ruby
require 'rails_helper'

 RSpec.describe "Posts", type: :request do
   ...
 end
```

- [ ] Inside this there should be two describe blocks one for `index` and another for `show` where you will test if the index or show template is rendering, if it includes the text you expect on the page, etc..

Hint:


```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
```